src/devices/src/virtio/queue.rs

```diff
@@ -93,7 +93,11 @@ impl<'a> DescriptorChain<'a> {
             Ok(ret) => ret,
             Err(err) => {
                 // TODO log address
-                error!("Failed to read virtio descriptor from memory: {}", err);
+                error!(
+                    "Failed to read virtio descriptor from memory at address 0x{:x}: {}",
+                    desc_head,
+                    err
+                );
```

Comment:
I think the arguments can be moved into the format string and from https://doc.rust-lang.org/std/fmt/trait.LowerHex.html it notes:

> The alternate flag, #, adds a 0x in front of the output.,

so we don't need to manually write the prefix `0x`

```suggestion
                error!("Failed to read virtio descriptor from memory at address {desc_head:#x}: {err}");
```